pyfr/plugins/ascent.py

```diff
@@ -0,0 +1,433 @@
+        pipel_cfg = {}
+        self.pipelines = pl = ConduitNode(self.conduit)
+
+        for k in self.cfg.items(cfgsect, prefix='pipeline-'):
+            pn = k.removeprefix('pipeline-')
+            cfg = self.cfg.getliteral(cfgsect, k)
+            pipel_cfg[pn] = cfg = [cfg] if isinstance(cfg, dict) else cfg
+
+            for j, filter in enumerate(cfg):
+                params = ConduitNode(self.conduit)
+
+                pl[f'pl_{pn}/f{j}/type'] = ptype = filter.pop('type')
+                for kf, vf in filter.items():
+                    if kf == 'output-name':
+                        if vf in self._fields_write:
+                            raise KeyError(f"Output name '{vf}' already used")
+                        self._fields_write.add(vf)
+                    elif kf == 'field':
+                        self._fields_read.add(vf)
+
+                    params[kf.replace('-', '_')] = vf
+                
+                if ptype in self.v_filter:
+                    self._fields_read.add('velocity')
+
+                pl[f'pl_{pn}/f{j}/params'] = params
+
+    def _init_scenes(self, cfgsect):
+        cons = self.cfg.items_as('constants', float)
+        self._scene_cfg = scene_cfg = {}
+        self.scenes = ConduitNode(self.conduit)
+
+        for k in self.cfg.items(cfgsect, prefix='scene-'):
+            sn = k.removeprefix('scene-')
+            scene_cfg[sn] = cfg = self.cfg.getliteral(cfgsect, k)
+
+            if cfg['type'] != 'mesh':
+                field = cfg['field']
+                self._fields_read.add(field)
+
+            for kc, vc in cfg.items():
+                if kc.startswith('field-expr'):
+                    self._add_field_expr(field, kc, vc, cons)
+                elif kc == 'pipeline':
+                    self.scenes[f's_{sn}/plots/p1/pipeline'] = f'pl_{vc}'
+                elif kc.startswith('render-'):
+                    rname = kc.removeprefix('render-')
+                    path = f's_{sn}/renders/r{rname}'
+
+                    self._render_options(path, vc)
+                elif kc != 'image-name':
+                    key = kc.replace('-', '_')
+                    self.scenes[f's_{sn}/plots/p1/{key}'] = vc
+
+            # If no render options then set image name
+            if not any(kc.startswith('render-') for kc in cfg):
+                gen = file_path_gen(self.basedir, cfg['image-name'], 
+                                    self.isrestart)
+                next(gen)
+                self._image_paths.append((f'scenes/s_{sn}/image_name', gen))
+
+    def _render_options(self, path, opts):
+        for k, v in opts.items():
+            if k != 'image-name':
+                # Replace for vectors
+                key = k.replace('_', '/')
+                # Replace for sections
+                key = key.replace('-', '_')
+                self.scenes[f'{path}/{key}'] = v
+
+        gen = file_path_gen(self.basedir, opts['image-name'], self.isrestart)
```

Comment:
Temped to rename `file_path_gen` to `_file_path_gen` and then introduce `file_path_gen` which gets the iterator, calls `next`, and then returns it.  Hides the `next` stuff from users.